test/http_responses_test.dart

```diff
@@ -9,63 +9,74 @@ import 'package:cronet/cronet.dart';
 import 'package:test/test.dart';
 
 void main() {
-  late HttpClient client;
-  late HttpServer server;
-  final sentData = 'Hello, world!';
-  setUp(() async {
-    client = HttpClient();
-    server = await HttpServer.bind(InternetAddress.anyIPv6, 5252);
```

Comment:
Instead of using so many different ports, which obviously increase the chances of failure, you should stick to using just 1. This is likely to happen on CI systems like Travis which would run on every single PR.

You can also simply set the port as `0`. See https://api.flutter.dev/flutter/dart-io/ServerSocket/bind.html#:~:text=version%206%20only-,If%20port%20has%20the%20value%200,The,-optional%20argument%20backlog
This way you don't have to bother about tests failing due to used up ports.

And the returned ServerSocket has the actual `port` which was assigned by the system.

Reply:
Ah.. I wasn't aware about this port `0` thing. I'll update it.

What port should I choose for that `wrong port` test?

Reply:
You can create an HttpServer with port 0, get the real port, and then close the socket.
This port should now definitely be closed. So you can use this.


Also you shouldn't use raw strings everywhere like this. You can make a `test_utils.dart` file to place constants (like HOST='localhost', PORT=0) and for functions which are used a lot throughout many tests, this improves code readability and changes to everything can be done in one single place.

Reply:
Okay, I'll do it :)